src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/AbstractEntityAIGuard.java

```diff
@@ -408,6 +430,27 @@ protected AIState attackPhysical()
         return GUARD_ATTACK_PHYSICAL;
     }
 
+    protected abstract int getAttackRange();
+
+    /**
+     * Gets the reload time for a Range guard attack.
+     *
+     * @return the reload time
+     */
+    protected int getAttackDelay()
+    {
+        if (worker.getCitizenData() != null)
+        {
+            return RANGED_ATTACK_DELAY_BASE / (worker.getCitizenData().getLevel() + 1);
+        }
+        return RANGED_ATTACK_DELAY_BASE;
+    }
+
+    private double getRandomPitch()
+    {
+        return PITCH_DIVIDER / (worker.getRNG().nextDouble() * PITCH_MULTIPLIER + BASE_PITCH);
+    }
+
     protected AIState attackRanged()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)